src/main/java/com/github/princesslana/eriscasper/data/util/Nullable.java

```diff
@@ -0,0 +1,89 @@
+package com.github.princesslana.eriscasper.data.util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class Nullable<T> {
+
+  private final Optional<T> value;
+
+  private Nullable(Optional<T> value) {
+    this.value = value;
+  }
+
+  public <U> Nullable<U> flatMap(Function<? super T, Nullable<U>> f) {
+    return isNull() ? Nullable.<U>ofNull() : f.apply(value.get());
+  }
+
+  public <U> Nullable<U> map(Function<? super T, ? extends U> f) {
+    return of(value.map(f));
+  }
+
+  public boolean isNull() {
+    return !value.isPresent();
+  }
+
+  public boolean isPresent() {
+    return value.isPresent();
+  }
+
+  public void ifPresent(Consumer<? super T> c) {
+    value.ifPresent(c);
+  }
+
+  @JsonValue
+  @javax.annotation.Nullable
+  public T get() {
+    return value.orElse(null);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this).add("value", get()).toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(value);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
+    }
+    if (obj == this) {
+      return true;
+    }
+    if (!(obj instanceof Nullable)) {
+      return false;
+    }
+    Nullable<?> rhs = (Nullable<?>) obj;
+
+    return Objects.equals(value, rhs.value);
+  }
+
+  private static <T> Nullable<T> of(Optional<T> value) {
+    return new Nullable<>(value);
+  }
+
+  public static <T> Nullable<T> ofNull() {
+    return of(Optional.empty());
```

Comment:
This might be better as a static constant. Just because having a null Nulllable is a constant value and is immutable. Doesn't make sense to create more and more. (it also is very common to use this.)

Reply:
Yeah, that's possible. I can't remember why I didn't. I seem to remember having an issue somewhere, but can't see where that would have been now.

Reply:
I don't really see any possible issue coming from it, Optional.empty() is constant afaik, and it being constant would only reduce method calls. Kind of nice to the stack and GC.